### PR TITLE
CrowdStrike bidirectional response actions (isolate & release)

### PR DESCRIPTION
**Pull request by joepeeples**
Contributes to https://github.com/elastic/security-docs/issues/5446 and https://github.com/elastic/security-docs-internal/issues/21 (internal).

### Previews

The format is a little different between serverless and stateful docs, due to each system's limitations. The serverless version uses a neat tabbed format to display one EDR system at a time, while stateful docs use separate sections or collapsible sections. The goal in both approaches is to avoid overwhelming the reader by limiting the amount of text displayed (one EDR system at at time).

Serverless docs:
- Click the [link below](https://github.com/elastic/security-docs/pull/5529#issuecomment-2215231221), then in the Security card click **View serverless docs**.
- Navigate to **Endpoint response actions**, then:
   - **Third-party response actions**
   - **Configure third-party response actions**

Stateful/ESS docs:
   - [Third-party response actions](https://security-docs_bk_5529.docs-preview.app.elstc.co/guide/en/security/master/third-party-actions.html)
   - [Configure third-party response actions](https://security-docs_bk_5529.docs-preview.app.elstc.co/guide/en/security/master/response-actions-config.html)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-5ffdt0rjp-elastic-dev.vercel.app

* Built with commit 19447e278ac5854dfe2122775b6890a64abe5c3e

> Issues? Visit #next-docs in Slack

File: docs/management/admin/response-actions-config.asciidoc
Comment (nastasha-solomon) [inline docs/management/admin/response-actions-config.asciidoc:86]: Same comment as above.

```suggestion
Create two API tokens in SentinelOne, and give them the minimum privilege required by the Elastic components that will use them:
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:24]: Same comment has [here](https://github.com/elastic/security-docs/pull/5529/files#diff-adbc9ab7e583838948471361b728951c75509c19855dc6520fa404ad0a73ae92R27).

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:37]: Optional edit

```suggestion
        - Give the API client the minimum privilege required to read CrowdStrike data and perform actions on enrolled hosts.
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:39]:
```suggestion
        - Take note of the client ID, client secret, and base URL; you'll need them in later steps when you configure ((elastic-sec)) components to access CrowdStrike.<br /><br />
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:84]: Optional edit 

```suggestion
        Create two API tokens in SentinelOne, and give them the minimum privilege required by the Elastic components that will use them:
```

File: docs/management/admin/response-actions-config.asciidoc
Comment (nastasha-solomon) [inline docs/management/admin/response-actions-config.asciidoc:74]: I think _in_ is more appropriate here since the menu is within the scope of the flyout.

```suggestion
This gives you visibility into CrowdStrike without needing to leave {elastic-sec}. You can perform supported endpoint response actions directly from alerts that the rule creates, by using the **Take action** menu in the alert details flyout.
```

File: docs/management/admin/response-actions-config.asciidoc
Comment (nastasha-solomon) [inline docs/management/admin/response-actions-config.asciidoc:129]:
```suggestion
This gives you visibility into SentinelOne without needing to leave {elastic-sec}. You can perform supported endpoint response actions directly from alerts that the rule creates, by using the **Take action** menu in the alert details flyout.
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:74]:
```suggestion
        This gives you visibility into CrowdStrike without needing to leave ((elastic-sec)). You can perform supported endpoint response actions directly from alerts that the rule creates, by using the **Take action** menu in the alert details flyout.
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (nastasha-solomon) [inline docs/serverless/endpoint-response-actions/response-actions-config.mdx:125]:
```suggestion
        This gives you visibility into SentinelOne without needing to leave ((elastic-sec)). You can perform supported endpoint response actions directly from alerts that the rule creates, by using the **Take action** menu in the alert details flyout.
```

**Review by nastasha-solomon — DISMISSED** [review]
Just left a handful of optional edits and some questions. Looks great overall though, and good use of the tabbed and collapse/expand components to organize information!

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [reply docs/management/admin/response-actions-config.asciidoc:27]:
```suggestion
* Endpoints must have actively running third-party agents installed.
```

File: docs/serverless/endpoint-response-actions/response-actions-config.mdx
Comment (joepeeples) [reply docs/serverless/endpoint-response-actions/response-actions-config.mdx:24]:
```suggestion
* Endpoints must have actively running third-party agents installed.
```

File: docs/management/admin/response-actions-config.asciidoc
Comment (paul-tavares) [inline docs/management/admin/response-actions-config.asciidoc:40]: Lets double check this step is correct, because this currently states that the 1 API key with full privileges to read and also execute response actions. This provides the integration policy with a token that IMO has too many unnecessary privileges.

File: docs/management/admin/response-actions-config.asciidoc
Comment (paul-tavares) [inline docs/management/admin/response-actions-config.asciidoc:43]: NOTE: there are two crowdstrike integrations. Lets confirm that the only needed one is this one you documented here (the other one is called 'CrowdStrike Falcon Intelligence')

File: docs/management/admin/response-actions-config.asciidoc
Comment (paul-tavares) [inline docs/management/admin/response-actions-config.asciidoc:70]: This, IMO, is not a good suggestion for the index name - it essentially says "create alerts for any type of crowdstrike data". I think (but you would need to confirm with @tomsonpl ) that there are specific indicates for crowdstrike alerts and other malware data, so perhaps using a more specific example would be best.

The other alternative would be to refer the customer tot he Integration overview page which describes the different types of data collected by the integration - that should give the user a good starting point on how to setup their Rule to only promote valid threats to SIEM alerts in kibana.

____

@caitlinbetz  - it would be great if we could get an example rule from one of our customers that run CrowdStrike just to see how they have setup their rule

**Review by paul-tavares — COMMENTED** [review]
Hi @joepeeples ,
I took a pass at these docs, but really I'm hopping that you can wait until sometime next week to get @tomsonpl 's review since he's the most knowledgeable with CrowdStrike.

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [reply docs/management/admin/response-actions-config.asciidoc:40]: Makes sense. The demo video and other materials I was able to find for CrowdStrike didn't specify multiple API clients, but we recommend 2 API keys for SentinelOne and could do something similar for CrowdStrike.

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [reply docs/management/admin/response-actions-config.asciidoc:70]: A sample index pattern or query would be preferable, I think — linking to [the integration's docs](https://www.elastic.co/docs/current/en/integrations/crowdstrike) could be overwhelming since it just lists every possible fields ingested, but doesn't recommend any particular pattern.

Is there an alert-specific pattern we could suggest, similar to how for SentinelOne we suggest `logs-sentinel_one.alert*`? That way the query only targets alerts, not any type of CrowdStrike data.

Comment (tomsonpl) [reply docs/management/admin/response-actions-config.asciidoc:70]: I would agree with @paul-tavares here, it's really up to user what query they want to use and most of the times they are pretty specific. For the docs I would suggest using something like this: 'use a query or index pattern that could find Crowdstrike data'   - of course in better English ;p 

WDYT @joepeeples ?

File: docs/management/admin/response-actions-config.asciidoc
Comment (tomsonpl) [reply docs/management/admin/response-actions-config.asciidoc:40]: Unfortunately I cannot confirm this - but you both may be right. Ingesting data might use a different clientId/secret than response actions - but it's rather Crowdstrike specific that I do not have information of ;/

**Review by tomsonpl — APPROVED** [review]
Added my 2 cents to Paul's great comments, but other than that - LGTM 👍 thank you!

**Review by caitlinbetz — APPROVED** [review]

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [reply docs/management/admin/response-actions-config.asciidoc:40]: OK, I can try to reword this to be a little less prescriptive about exact number of API clients, give more general guidelines and leave it up to the user's discretion.

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [reply docs/management/admin/response-actions-config.asciidoc:70]: Similar to above, I'll reword this to be more general guidance, and probably link to our CrowdStrike integration docs page since it lists all the available fields. Even if it's up to the user to write queries however they see fit, it'd be ideal if we could at least give some example queries to illustrate how to detect specific things, but maybe we can add that in a later revision once we know more about the data.

**Comment by joepeeples** [comment]
@paul-tavares @tomsonpl I pushed a few revision commits to address your feedback above (API client privs, rule query suggestions). Please check out the updated [preview page](https://security-docs_bk_5529.docs-preview.app.elstc.co/guide/en/security/master/response-actions-config.html) and let me know if this works. We can always add more details later to fine-tune our guidance for customers. Thanks!

(FYI, I haven't updated the serverless side yet, will push one commit once we nail down the content in ESS, since that's easier to preview.)

File: docs/management/admin/response-actions-config.asciidoc
Comment (joepeeples) [inline docs/management/admin/response-actions-config.asciidoc:66]: Wondering if the API client used to perform actions _also_ needs privs to read CrowdStrike data?

**Review by paul-tavares — APPROVED** [review]

**Review by nastasha-solomon — APPROVED** [review]

File: docs/management/admin/response-actions-config.asciidoc
Comment (tomsonpl) [reply docs/management/admin/response-actions-config.asciidoc:66]: 🤷 no idea, but I would assume it's nicely explained in CrowdStrike when creating credentials, so imho the users shouldn't be confused by this 👍

**Review by tomsonpl — APPROVED** [review]
LGTM 👍  thank you !

**Pull request merged**
